sdk/search/azure-search/src/main/java/com/azure/search/documents/implementation/util/CustomPatternAnalyzerSerializer.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.implementation.util;
+
+import com.azure.search.documents.models.PatternAnalyzer;
+import com.azure.search.documents.models.PatternTokenizer;
+import com.azure.search.documents.models.RegexFlags;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
+import java.io.IOException;
+import java.util.stream.Collectors;
+
+/**
+ * Custom serializer for {@link PatternAnalyzer}, which flatten the list of {@link RegexFlags}.
+ */
+public class CustomPatternAnalyzerSerializer extends JsonSerializer<PatternAnalyzer> {
+    private static final String DELIMITER = "|";
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param analyzer The {@link PatternAnalyzer} needs to serialize
+     * @param gen Generator used to output resulting Json content
+     * @param serializers Provider that can be used to get serializers for
+     *   serializing Objects value contains, if any.
+     * @param typeSer Type serializer to use for including type information
+     * @throws IOException If IO operation error occurs.
+     */
+    @Override
+    public void serializeWithType(PatternAnalyzer analyzer, JsonGenerator gen, SerializerProvider serializers,
+        TypeSerializer typeSer) throws IOException {
+        gen.writeStartObject();
+        gen.writeStringField(typeSer.getPropertyName(), typeSer.getTypeIdResolver().idFromValue(analyzer));
+        serialize(analyzer, gen, serializers);
+        gen.writeEndObject();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param analyzer The {@link PatternAnalyzer} needs to serialize
+     * @param jsonGenerator Generator used to output resulting Json content
+     * @param serializerProvider Provider that can be used to get serializers for
+     *   serializing Objects value contains, if any.
+     * @throws IOException If IO operation error occurs.
+     */
+    @Override
+    public void serialize(final PatternAnalyzer analyzer, final JsonGenerator jsonGenerator,
+        final SerializerProvider serializerProvider) throws IOException {
+        if (analyzer == null) {
+            return;
+        }
```

Comment:
What is the output when analyzer is null?

Reply:
> My main concern is around the serializers and deserializers and some corner cases they may have, could you add tests which specifically target them? Make sure to add guards even in scenarios where the service may not have those returns now since it is always possible to change in the future, especially around default JSON values.

Totally agreed for adding more tests to cover more corner scenarios. Will keep working on this while reviewing the body.

I have several test cases change to cover the major workflow.

Reply:
I cannot have the case. `Index.setAnalyzer` takes list<Analyzer>

There is no way to have a null PatternAnalyzer pass in. 

I plan to remove it.